x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/amazonbedrock/embeddings/AmazonBedrockEmbeddingsServiceSettings.java

```diff
@@ -0,0 +1,214 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.services.amazonbedrock.embeddings;
+
+import org.elasticsearch.common.ValidationException;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.core.Nullable;
+import org.elasticsearch.index.mapper.vectors.DenseVectorFieldMapper;
+import org.elasticsearch.inference.ModelConfigurations;
+import org.elasticsearch.inference.SimilarityMeasure;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xpack.inference.services.ConfigurationParseContext;
+import org.elasticsearch.xpack.inference.services.ServiceUtils;
+import org.elasticsearch.xpack.inference.services.amazonbedrock.AmazonBedrockProvider;
+import org.elasticsearch.xpack.inference.services.amazonbedrock.AmazonBedrockServiceSettings;
+import org.elasticsearch.xpack.inference.services.settings.RateLimitSettings;
+
+import java.io.IOException;
+import java.util.Map;
+import java.util.Objects;
+
+import static org.elasticsearch.xpack.inference.services.ServiceFields.DIMENSIONS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.MAX_INPUT_TOKENS;
+import static org.elasticsearch.xpack.inference.services.ServiceFields.SIMILARITY;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalBoolean;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractOptionalPositiveInteger;
+import static org.elasticsearch.xpack.inference.services.ServiceUtils.extractSimilarity;
+
+public class AmazonBedrockEmbeddingsServiceSettings extends AmazonBedrockServiceSettings {
+    public static final String NAME = "amazon_bedrock_embeddings_service_settings";
+    static final String DIMENSIONS_SET_BY_USER = "dimensions_set_by_user";
+
+    private final Integer dimensions;
+    private final Boolean dimensionsSetByUser;
```

Comment:
I might have missed it but do we or do we plan on sending this to an external provider? I think I only saw that we use it in the `updateModelWithEmbeddingDetails()` method. If we don't send it to an external provider then the user set dimensions might mess the semantic text field 🤔 . What I mean is, if a users sets it to 5 but the actual stored embeddings is 1000 I'm not sure what happens.

I think my vote would be to remove this if we don't plan on using it in the near future. If we do plan on using it soon, we might want to prevent a user from setting it for now 🤷‍♂️

Reply:
I think this is an oversight on my part... for [Amazon Titan embeddings](https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-titan-embed-text.html) - you _can_ pass the dimensions, but [Cohere](https://docs.aws.amazon.com/bedrock/latest/userguide/model-parameters-embed.html) does not have this... I'll fix this up, with some validations on this.

Reply:
Ah ok cool 👍 yeah don't forget to pass it along in the titan request, or at least I missed where we are doing that.

Reply:
Sooooo - here's something silly... Amazon Bedrock G1 models don't allow the `dimensions` parameter, but the version 2 does..:

```
Titan Embeddings G1 - Text doesn't support the use of inference parameters. The following sections detail the request and response formats and provides a code example.
```

I think for now, I'll omit the dimensions all together because there's no definitive way we can tell which model the end user is using easily (we _could_ check on the model ID as the string to see if it matches the v2 version, but, if they are using a custom ARN, it'd require a lot more hoops to jump through that I don't think is worth it at this point in time...)

Reply:
>  If we do plan on using it soon, we might want to prevent a user from setting it for now

I'll go this route for now as it's probably the safest.

Reply:
> I think for now, I'll omit the dimensions all together because there's no definitive way we can tell which model the end user is using easily

👍 sounds good